[EC-457] Component library icon buttons

coroiu opened the pull request:
## Type of change

- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Implement new component: Icon Button

## Code changes

- **components/src/banner:**
    - Replace close button with new icon button (and add `aria-label`)
    - Add support for tracking `onClick` events in storybook using actions
- **components/src/button:**
    - Make `hover` state independent of `focus` state
- **components/src/dialog:**
    - Replace close button with new icon button (and add `aria-label`)
    - Add example with trash can icon in modal footer
- **components/src/form-field:**
    - Remove `I18nPipe`
- **components/src/icon-button:**
    - Main topic of this PR
    - Uses `box-shadow` focus ring workaround
    - Standalone (does not use button)
- **components/src/shared:**
    - Add new shared module with `I18nPipe` 
    - Needed because `aria-label` is used in Banner and Dialog to describe Close button
- **components/src/tw-theme.css**:
    - Added transparent hover background for shadow variants of icon buttons.

## Screenshots

See Storybook

## Before you submit

- [ ] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

coroiu commented on libs/components/src/icon-button/icon-button.component.ts:
```diff
@@ -0,0 +1,121 @@
+  default: [],
+};
+
+@Component({
```
**point-of-interest:** This component does not re-use the regular button component. I'm not sure if the complexity added by extending/somehow combining the buttons is worth getting rid of some of the code duplication. Thoughts?

coroiu replied:
That's exactly how I thought about it, thanks for validating!

coroiu commented on libs/components/src/icon-button/icons.ts:
```diff
@@ -0,0 +1,63 @@
+export const IconSvg = {
```
**issue:** copy-pasta, remove

coroiu commented on libs/components/src/tw-theme.css:
```diff
@@ -1,4 +1,6 @@
 :root {
+  --color-transparent-hover: rgb(0 0 0 / 0.03);
```
**point-of-interest:** This variable name diverges a bit from the rest by containing some information about how it's supposed to be used. The other variables are just generic colors without any information about usage

danielleflinn commented:
@coroiu I noticed the following divergence from Figma:
• Some focus states show up as white when they should be either` text-main`, `text-contrast`, or `primary-700` depending on the style (see Figma)
• Primary includes a `primary-500` outline on the hover state that was not expected
• For icon buttons, the padding should be consistent for x and y axis's so the resulting button is square (this is different from the button component styles where there is less padding on the top

I know this is a draft so, ignore this comment if you are already aware and addressing these...

coroiu commented:
@danielleflinn Thank you for the review! It was a draft but I had missed some of the things you mentioned!

- _Some focus states show up as white when they should be either text-main, text-contrast, or primary-700 depending on the style (see Figma)_
  Which parts are you referring to? Icon font color? Background? I don't see any differences but it's probably because I've stared at them for too long 😅 
- _For icon buttons, the padding should be consistent for x and y axis's so the resulting button is square (this is different from the button component styles where there is less padding on the top_
 The buttons should be square now (equal padding on all sides). However the `default` size is quite tricky due to different `line-height` on Button and IconButton. I'm interpreting the intention of the design to be that buttons and icon buttons should have the same height (when using `default` size) and that icon buttons should always be square. Is that correct?
- _Primary includes a primary-500 outline on the hover state that was not expected_
  Fixed!

danielleflinn commented:
@coroiu Focus states colors look accurate now. Previously, the focus state was taking on a white color in instances where it should have been main or primary. They all look correct now.

Maybe we should update these to` focus-visable`? I think there was a discussion on slack to always use` focus-visable `as the default component focus. 

As far as padding on default; you are correct in assuming we want the height to be the same for default as the button; I thought I'd done the math so` tw-p-3` would work. But looking at the dialog it looks like` tw-p-2.5` will get us to almost the same height (~1px off). Thoughts on using `p-2.5`? 

I'm going to share this with the other designers and see if there is in final feedback from a design perspective. Thanks for your work on these, and for updating the banner and dialogs components :)

coroiu commented:
> @coroiu Focus states colors look accurate now. Previously, the focus state was taking on a white color in instances where it should have been main or primary. They all look correct now.

Awesome! 

> Maybe we should update these to` focus-visable`? I think there was a discussion on slack to always use`focus-visable`as the default component focus.

Yeah I've been part of that as well, I'll update it (and update the regular buttons as well)

> As far as padding on default; you are correct in assuming we want the height to be the same for default as the button; I thought I'd done the math so` tw-p-3` would work. But looking at the dialog it looks like` tw-p-2.5` will get us to almost the same height (~1px off). Thoughts on using `p-2.5`?

Yeah `2.5` gets us most of the way but it's still a bit off. I'll try to look into it little bit more, worst case I'll leave it at `2.5`.

> I'm going to share this with the other designers and see if there is in final feedback from a design perspective. Thanks for your work on these, and for updating the banner and dialogs components :)

Awesome thank you! I'll mark the PR as ready for review to get some technical comments on it as well

eliykat commented on libs/components/src/icon-button/icon-button.component.ts:
```diff
@@ -0,0 +1,122 @@
+    "hover:tw-border-text-contrast",
+    "focus:before:tw-ring-text-contrast",
+    "disabled:hover:tw-bg-transparent",
+    "disabled:hover:tw-border-transparent",
```
`disabled:hover:tw-border-transparent` looks like it's common to all styles and could be moved to the `classList` method.

eliykat commented on libs/components/src/icon-button/icon-button.component.ts:
```diff
@@ -0,0 +1,122 @@
+const styles: Record<IconButtonStyle, string[]> = {
+  contrast: [
+    "tw-bg-transparent",
+    "!tw-text-contrast",
```
Why do text stylings need to be `!important`?

coroiu replied:
Yeah this bothers me as well... It's because bootstrap/our css overrides it in the web vault

Hinton replied:
There is a tech debt task to clear that up after we remove bootstrap

eliykat replied:
Sounds good, thanks for the clarification!

eliykat reviewed (changes requested):
Mostly looks good to me, just a few comments below.

I feel obliged to mention that you could also write tests, but to be honest I'm not sure what's worth testing here given that the icon doesn't have particularly complex behaviour.

coroiu commented on libs/components/src/icon-button/icon-button.component.ts:
```diff
@@ -0,0 +1,122 @@
+  ],
+};
+
+export type IconButtonSize = "base" | "xl" | "2xl" | "default";
```
@eliykat The names `base`, `xl`, `2xl` come directly from tailwind, so in the context of trying to match an icon button to surrounding text keeping the names would probably simplify that process. Do you think simply changing `default` to something else would be enough? Do you have any name suggestions for `looks like a button` 😅 ? Maybe `regular`, `large`, `xlarge`, `looks like a button`?

@danielleflinn What do you think about this? The design document only refers to these using the underlying values, i.e. `1 rem`, `1.25 rem`, `1.5 rem` and `default`. I'm guessing the fact that `default` is a variation of `main` with different padding might add to the confusion slightly.

danielleflinn replied:
I think keeping the names similar to the tailwind text sizes will help reduce confusion because the non-default variants are often used near text. So an engineer could think "Oh the dialog heading uses text-lg, I should be using the large icon button for close."

So with that in mind, maybe `base`, `large`, `xlarge`, and 'button-like'.

Another thought: use the above name but move the 'button-like' variant form being a variant of the icon button and make it a variant for the 'button' component. Here we would name it 'icon-only' or something and only support the contained/outlined styles. Not sure what would be the preference from an implementation perspective...

eliykat replied:
I didn't realise it tracks Tailwind sizing - taking a closer look, that makes sense.

I have to admit I still don't understand `default` though (and you're right, this is what confused me). It doesn't have any `tw-text-[size]` class, so is it intended to match the surrounding text size? If it only has padding and no change in text size, isn't that a style and not a size? But then if we make it a style (as @danielleflinn mentions), it means that you can't combine it with any other styles, which seems to be the intention.

In what circumstances do we use the `default` size? `looks like a button` - don't all of these look like buttons? 😂

eliykat replied:
Assuming that my understanding above is basically correct, maybe an additional `input`: `export type IconButtonPadding = "none" | "button"`? (and it defaults to `button`)

eliykat replied:
Assuming that my understanding above is basically correct, I don't know if I have a better solution. Changing it to `button-like` is more confusing to me, it makes it even clearer that it's not a size. Making it a style makes  sense, but you can't combine styles. You could break out `padding` into its own input, but then you're losing the rigid presets we want in the component library. 🤷 I might be overthinking it.

coroiu replied:
I really like when people express these kind of raw thought processes! Here are some of my thoughts:

> I might be overthinking it.

I just wanted to bring this one up first because I've had the exact same thoughts and I'm still not 100% convinced regarding the implementation. I don't think you're overthinking it at all.

> `default` [...] doesn't have any tw-text-[size] class, so is it intended to match the surrounding text size?

No, my fault! I left it out because I was confused by `default` at the time. The text size should match `bitButton`.

> If it only has padding and no change in text size, isn't that a style and not a size? 

I would argue "no" because `size` refers to the size of the entire button, not the `font-size` within. Changing the padding changes the size. Furthermore `style` is related to themes and colors and does not impact size.

> The main use case for default seems to be that it gives you padding that is consistent with bitButton, is that what you mean by looks like a button? (All of these look like buttons to me! lol)

This is correct!

> You could break out padding into its own input, but then you're losing the rigid presets we want in the component library. 

Yeah and then we would be able to create icon buttons that are not present in the design documents. We could add runtime checks for certain combinations, but then we'd confuse the developer by making it unclear which combinations are allowed.

#### And here is one thought I've had myself:
When I started out implementing these icon buttons I had a big choice to make: Do I use the `bitButton` component (actually directive but let's ignore that) in some way, or do I create an entirely new standalone component? I chose the latter.

I argued that re-using `bitButton` would mean introducing the `size` concept and the additional styles to `bitButton`. That would in turn enable developers to create `bitButton` variations that the design documents do not describe or support. Using `Component` instead of `Directive` is also a conscious choice - it prohibits any other content inside the icon button, while `bitButton` is a lot more open. For me this was the deal breaker.

I'm not married to this conclusion though, and for me the important part is making these components easy to use and understand. "Easy to use" meaning **easy to do the right thing** and **hard to do the wrong thing**. The underlying implementation comes in (close) second place. Given this, do you think anything should be done differently? Would you/developers have an easier time understanding how to use this component if we re-used the `bitButton` component instead?

coroiu replied:
I really like when people express these kind of raw thought processes! Here are some of my thoughts on what you wrote:

> I might be overthinking it.

I just wanted to bring this one up first because I've had the exact same thoughts and I'm still not 100% convinced regarding the implementation. I don't think you're overthinking it at all.

> `default` [...] doesn't have any tw-text-[size] class, so is it intended to match the surrounding text size?

No, my fault! I left it out because I was confused by `default` at the time. The text size should match `bitButton`.

> If it only has padding and no change in text size, isn't that a style and not a size? 

I would argue "no" because `size` refers to the size of the entire button, not the `font-size` within. Changing the padding changes the size. Furthermore `style` is related to themes and colors and does not impact size.

> The main use case for default seems to be that it gives you padding that is consistent with bitButton, is that what you mean by looks like a button? (All of these look like buttons to me! lol)

This is correct! But when I say `looks like a button` I specifically mean our own button, i.e. `looks like a bitButton`. These two should look the same: `<button bitIconButton="bwi-trash" size="default">`, `<button bitButton><i class="bwi-trash"></i></button>`. This is what I'm most unsure about though. Maybe this is confusing. We can do the same thing in different ways, and the difference is unclear. 

@danielleflinn Are the combinations `[style=contrast size=default]` `[style=main size=default]` `[style=muted size=default]` something that we expect to use? If not then I'm leaning towards removing `size=default` as an option from `bitIconButton` completely, and use the regular `bitButton` in those cases.

> You could break out padding into its own input, but then you're losing the rigid presets we want in the component library. 

Yeah and then we would be able to create icon buttons that are not present in the design documents. We could add runtime checks for certain combinations, but then we'd confuse the developer by making it unclear which combinations are allowed.

#### And here is one thought I've had myself:
When I started out implementing these icon buttons I had a big choice to make: Do I use the `bitButton` component (actually directive but let's ignore that) in some way, or do I create an entirely new standalone component? I chose the latter.

I argued that re-using `bitButton` would mean introducing the `size` concept and the additional styles to `bitButton`. That would in turn enable developers to create `bitButton` variations that the design documents do not describe or support. Using `Component` instead of `Directive` is also a conscious choice - it prohibits any other content inside the icon button, while `bitButton` is a lot more open. For me this was the deal breaker.

I'm not married to this conclusion though, and for me the important part is making these components easy to use and understand. "Easy to use" meaning **easy to do the right thing** and **hard to do the wrong thing**. The underlying implementation comes in (close) second place. Given this, do you think anything should be done differently? Would you/developers have an easier time understanding how to use this component if we re-used the `bitButton` component instead?

danielleflinn replied:
@coroiu I can't think of anywhere the `[style=contrast size=default]` `[style=main size=default]` `[style=muted size=default] `are used in the product today. The desire from design is that if a regular` bitButton` is used for just an icon, the full container should be square (currently buttons on the web vault today with just icon appear as rectangles). Is it possible to use the regular bitButton and adjust padding to be square if only the icon is present?

eliykat replied:
> I can't think of anywhere the [style=contrast size=default] [style=main size=default] [style=muted size=default] are used in the product today [...]

Ok, so at the very least we still need to be able to combine `size=default` with `style=primary`, `style=secondary` and `style=danger`. So the `size=default` classes definitely can't be a style.

> When I started out implementing these icon buttons I had a big choice to make: Do I use the bitButton component (actually directive but let's ignore that) in some way, or do I create an entirely new standalone component? I chose the latter. [...]

I agree with your thoughts here, I think trying to go back and combine it with `bitButton` in some or all cases is only going to create more problems, particularly given design requirements that an icon button is square (unlike other `bitButton` cases).

> [...] size refers to the size of the entire button, not the font-size within. Changing the padding changes the size. Furthermore style is related to themes and colors and does not impact size.

I think this is our answer. If we think of sizing in this way, then it makes sense for this to be `button` sizing (i.e. `size="button"`). It's not perfect, because _all these options are buttons_, but the _sizing_ reflects the sizing of our component library button specifically.

So where I land on this is: call this `button` sizing and also add a jsdoc-style comment to explain more precisely what we mean by this. e.g. 

```ts
// "button" has similar sizing/padding as a bitButton, but used where it's an icon only. Use this instead of placing an icon inside of a bitButton.
button: ["tw-px-2.5", ...]
```

Arguably it could be `sizing="bitButton" but I think that's confusing in a different way 😅

coroiu replied:
I agree, `button` is a better name, I'll update the name, add the comment, but keep the rest of the implementation the same.

coroiu commented on libs/components/src/shared/i18n.pipe.ts:
```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from "@angular/core";
```
@Hinton was the one that duplicated it in the first place so this was double checked with him before moving this to a new `shared` module. That said, I think it's good that you're bringing this up. I assume that this depends on how we choose to set up our dependencies so we don't get circular dependencies between packages. I.e. We have to duplicate the pipe if we ever want `JslibModule` to be able to depend on the Component Library. Or maybe we are trying to deprecate parts of `JslibModule` and move to the Component Library? I'm not yet familiar with the thinking behind these modules so I'm probably not qualified to answer these things 🙂

Hinton replied:
libs/angular should depend on libs/components, but it's only used in web which makes this a bit tricky. I believe webpack tree-shaking would correctly resolve this but haven't had time to investigate. The libs/angular is also quite big and could benefit from being broken down.

As a quick work-around I copied the pipe to components.

Hinton replied:
Can we also re-add the comment that this is a work-around?

eliykat replied:
> We're only using the pipe in stories.

Not sure if this is what you mean, but the reason I raised it is because this pipe is used in the components themselves and will be used in production. `SharedModule` is imported by the module for each component (e.g. `banner.module.ts`), which is in turn imported by the client production code. It's not like a service where the parent injector will override it. I think `story-utils` might be misleading.

That said, this does all sound like a larger problem, I'd be happy with just restoring the comment that it's a work-around. And maybe considering creating a tech debt ticket to tackle it at some stage.

coroiu commented on libs/components/src/icon-button/icon-button.component.ts:
```diff
@@ -0,0 +1,122 @@
+
+      // Workaround for box-shadow with transparent offset issue:
+      // https://github.com/tailwindlabs/tailwindcss/issues/3595
+      // Switch to outline with outline-offset when Safari supports border radius on outline.
```
Ah, yeah...

So `focus:before:tw-ring` is a workaround for `focus:tw-ring` which in turn (I would argue) is a workaround for `tw-outline`. It is this way becuase:

- `tw-outline` (or really just `outline: 2px;`) doesn't support border radius on Safari
- `focus:tw-ring` (or "fancy box shadow magic") has the weird bleed issue

So I mention Safari because that's what we ideally should be using everywhere. Do you think I should try to clarify this in some way?

coroiu replied:
I expanded the comment a bit so it says the following:

> // Workaround for box-shadow with transparent offset issue:
> // `https://github.com/tailwindlabs/tailwindcss/issues/3595`
> // Remove `before:` and use regular `tw-ring` when browser no longer has bug, or better:
> // switch to `outline` with `outline-offset` when Safari supports border radius on outline.
> // Using `box-shadow` to create outlines is a hack and as such `outline` should be preferred.

eliykat replied:
Yes, perfect explanation in the updated comment!

coroiu commented:
@danielleflinn I have made the following changes

- I fixed the `default` icon sizing using the you suggestion with untouched `line-height`, `px-1.5` and `py-2.5`. I really liked this approach because it's the exact same way the button gets its own height.
- Changed to `focus-visible` for both icon button and regular button.
- Added `title` and `aria-label` to where it was missing, including the storybook templates to follow the rule. 
   I considered adding automatic attribute mirroring (so one only needed to enter one of them), but eventually decided not to proceed with it. Seemed unnecessarily complicated 🤷‍♂️

Hinton commented:
![image](https://user-images.githubusercontent.com/137855/187881147-cc0a3021-1391-46d3-a624-886743cda5a4.png)
Dialogs looks a bit off

eliykat reviewed (commented):
LGTM pending resolution of the few discussion threads above.

coroiu commented:
> @Hinton 
> ![image](https://user-images.githubusercontent.com/137855/187881147-cc0a3021-1391-46d3-a624-886743cda5a4.png) Dialogs looks a bit off

Regarding the buttons. I agree that they might benefit from a bit more padding. I've followed the written rules in figma (quote: `the padding on the rem sizes uses .125rem on all sides`). However after looking closer at the examples in figma, they do seem to have a bit more padding.

Figma:
![image](https://user-images.githubusercontent.com/2285588/188153558-8cd29222-1573-4e4e-bf77-0290bf14f42a.png)

Browser:
![image](https://user-images.githubusercontent.com/2285588/188153700-d6793fe7-fbfc-419a-bc5a-523c8d96ecf4.png)

Comparison between figma and browser is quite hard because they don't use the same root font size (figma is 14px, web is 16px).
**However** we can se that the figma icon is around 14x14px with a padding of ~3.5-4px, while the browser icon is around 16x16px ~2px. The icon sizes seem to be correct, but the padding figma padding does not seem to be 0.125rem which would be 1.75px. It seems to use 0.25rem. @danielleflinn am I on the right track here?

coroiu commented:
@Hinton @danielleflinn what do you think about doubling padding from 0.125rem to 0.25rem? I'm leaning towards 0.25rem

<img align="left" width="350" alt="image" src="https://user-images.githubusercontent.com/2285588/188414377-ef6d6e1b-0997-493b-b949-ec7d006c14a2.png">
<img align="left" width="350" alt="image" src="https://user-images.githubusercontent.com/2285588/188414484-751c0c41-2742-4051-bd85-813d10f8a98b.png">

Hinton commented:
@coroiu how does that look for other icons?

jtouchstonebw commented:
> @Hinton @danielleflinn what do you think about doubling padding from 0.125rem to 0.25rem? I'm leaning towards 0.25rem
> 
> <img alt="image" width="350" src="https://user-images.githubusercontent.com/2285588/188414377-ef6d6e1b-0997-493b-b949-ec7d006c14a2.png">
> 
> <img alt="image" width="350" src="https://user-images.githubusercontent.com/2285588/188414484-751c0c41-2742-4051-bd85-813d10f8a98b.png">

Is there a reason that the close icon button can't follow the same sizing conventions as the delete (trash) icon button, minus the red/danger color? Would it simplify things to use the same styling for icon buttons in modals?

danielleflinn commented:
> @Hinton @danielleflinn what do you think about doubling padding from 0.125rem to 0.25rem? I'm leaning towards 0.25rem

I'm good with using the .25rem padding. 

> Is there a reason that the close icon button can't follow the same sizing conventions as the delete (trash) icon button, minus the red/danger color and border? Would it simplify things to use the same styling for icon buttons in modals?

@jtouchstonebw No reason, we were starting with mimicking the existing modal styles which use a smaller icon for close. 

@coroiu can you try a version using the `size-button` variant for close in the dialog? Also I noticed in the most recent storybook build the delete icon button is sized a bit funky. I think it is because it is still using the `size=default`.

coroiu commented:
> Also I noticed in the most recent storybook build the delete icon button is sized a bit funky. 

Good catch, fixed!

> can you try a version using the size-button variant for close in the dialog? 
@danielleflinn absolutely, here are some comparisons, let me know if you want an interactive example published to chromatic

![image](https://user-images.githubusercontent.com/2285588/189060674-370efddd-f381-40f9-bdcb-fe6ac76192f6.png)

![image](https://user-images.githubusercontent.com/2285588/189059968-33681e55-a7b0-4292-b610-1ddedf14bab1.png)

danielleflinn commented:
@coroiu that looks great! Design is also good with the added height the larger size brings to the header. 

In light of changing the dialog's close button to the icon-button `button` variant and the discussion above about the naming convention. Design has re-evaluated the need for the xl and 2xl variants. The thought is to have only 2 variants for size: `small` (the current `base` variant), and `default` (the current `button` variant).

- `default` will be used for the close buttons in both the dialog and banner, and as needed throughout the product. 
- `small` will be used for the expand/collapse button for collection/folder trees and other areas a smaller icon may be necessary. 

We are hoping this will help both with maintaining this component in the long run and knowing when to use which size during development. @coroiu thoughts on the above proposal? 

If we do move ahead with the above, let's adjust the right padding on the banner to 1rem so when the close button is hovered/focused the surrounding padding is even. (see Figma)

coroiu commented:
@danielleflinn I think that sounds great! I've updated so that we now only have `default` and `small` and I've updated the Dialog and Banner to use `default`.

> If we do move ahead with the above, let's adjust the right padding on the banner to 1rem so when the close button is hovered/focused the surrounding padding is even. (see Figma)

Regarding the padding, I set it to `0.625rem` to match the top/bottom padding, it now has equal spacing on all sides :)

danielleflinn reviewed (changes requested):
@coroiu this looks great! I noticed two areas that should be fixed before merging: 
The Simple dialog "Custom Icon" story lost the `bwi-star` + `success-500` that previously was in that story ([see Chromatic UI Reviews](https://www.chromatic.com/test?appId=62a88a6de5b807fa98886113&id=631f470b571a9478b34d0a85)). 

It also looks like some of the behavior added from @Hinton 's [Simple Dialog width](https://github.com/bitwarden/clients/pull/3475) PR is not present in this work. I'm not sure if this branch needs to be updated from master since that was just merged or if this is caused by something else. 

Note: reminder to review and approve the latest UI Tests in Chromatic as outlined in the updated [CL Chromatic process](https://bitwarden.atlassian.net/wiki/spaces/EN/pages/245071877/Chromatic)

coroiu commented:
> The Simple dialog "Custom Icon" story lost the bwi-star + success-500 that previously was in that story ([see Chromatic UI Reviews](https://www.chromatic.com/test?appId=62a88a6de5b807fa98886113&id=631f470b571a9478b34d0a85)).

My bad! I got some merge conflict and this must've slipped through, thanks for noticing! I've approved the UI review now and I think everything should be in order!

> It also looks like some of the behavior added from @Hinton 's https://github.com/bitwarden/clients/pull/3475 PR is not present in this work. I'm not sure if this branch needs to be updated from master since that was just merged or if this is caused by something else.

I merged the latest changes and everything seems to be included now 🙂

danielleflinn reviewed (approved):
Looks great! Thanks @coroiu for all your work on this.

eliykat reviewed (changes requested):
Looking good, I especially like simplifying the available size options. Just a few small questions before we merge.

coroiu commented on libs/components/src/dialog/dialog/dialog.stories.ts:
```diff
@@ -46,9 +50,16 @@ const Template: Story<DialogComponent> = (args: DialogComponent) => ({
+        bitIconButton="bwi-trash"
+        buttonType="danger"
+        size="default"
+        title="Delete"
+        aria-label="Delete"></button>
```
The story templates are really just examples so I don't really think it matters either way 🤷 The other stories seem to hard-code strings quite often as well so I don't think this is against convention

coroiu commented on libs/components/src/dialog/simple-dialog.service.stories.ts:
```diff
@@ -60,13 +66,24 @@ export default {
   decorators: [
     moduleMetadata({
       declarations: [
+        DialogComponent,
```
Removed!

coroiu commented on libs/components/src/dialog/simple-dialog/simple-dialog.stories.ts:
```diff
@@ -10,7 +11,7 @@ export default {
   decorators: [
     moduleMetadata({
-      imports: [ButtonModule],
+      imports: [ButtonModule, SharedModule],
```
Removed!

coroiu commented on libs/components/src/dialog/dialog/dialog.stories.ts:
```diff
@@ -38,6 +39,9 @@ export default {
   },
+  argTypes: {
+    onClose: { action: "onClose" },
+  },
```
I removed this completely and described why in the commit message:

> Removing this storybook action because we already test it as part of the dialog service stories.
> It requries mocking the dialogRef to capture the close function which makes this story more complex    
> but adds very little value as we already test it elsewhere.

eliykat reviewed (approved):
Good work on this @coroiu !
